lib/src/infrasructure/repositry/auth_repositry/auth_repositry.dart

```diff
@@ -88,83 +89,45 @@ class AuthRepository implements IAuthRepository {
+  //   try {
+  //     SharedPreferences prefs = await SharedPreferences.getInstance();
+  //     prefs.setString('access_token', accessToken);
+  //     print('Access token saved: $accessToken');
+  //   } catch (e) {
+  //     print('Error accessing SharedPreferences: $e');
+  //   }
+  // }
 
   @override
   Future<AppUser> checkAuth() async {
     try {
-      SharedPreferences prefs = await SharedPreferences.getInstance();
-      if (prefs.getString('access_token') == null) {
-        log("Invalid Token");
-        throw Exception("Invalid Token");
-      }
-      // Implimented Intersepter ....
-      var dio = Dio();
-      dio.interceptors.add(Dionterceptor());
-      final response = await dio.get(ApiEndpoints.checkAuth, options: Options(headers: {"Authorization": "Bearer ${prefs.getString('access_token')}"}));
-      log(prefs.getString('access_token').toString());
-      log("Token: ${prefs.getString('access_token')}");
-      log("Request URL: ${ApiEndpoints.checkAuth}");
-      log("Request Headers: ${response.data.headers}");
-      log("Response status code: ${response.statusCode}");
-      log("Response body: ${response.data}");
+      final response = await api.profile.get(
+        ApiEndpoints.checkAuth,
+      );
+
+      final AppUser user = AppUser.fromJson(response.data);
 
-      if (response.statusCode == 200) {
-        final AppUser user = AppUser.fromJson(response.data);
-        print(user);
-        print("checkAuth Success");
-        return user;
-      } else {
-        throw Exception('Failed to check authentication..');
-      }
+      return user;
     } catch (error) {
-      print('Error: $error');
-      throw Exception('Failed to check authentication..');
+      throw Exception(error);
     }
   }
-
   @override
-  Future<AppUser> verifyOtp({required String otp, required String userId}) async {
-    log(userId, name: "User id From  Repo");
-    log(otp, name: "Otp id From  Repo");
+  Future<AppUser> verifyOtp(
+      {required String otp, required String userId}) async {
     try {
-      final Map<String, dynamic> data = {
+      final Map<String, dynamic> requestData = {
         'otp': otp,
         'user_id': userId,
       };
-      var dio = Dio();
-      dio.interceptors.add(Dionterceptor());
-      final response = await dio.post(ApiEndpoints.verifyOtpApi,
-          options: Options(
-            headers: {"Tokenvalid": AppConstants.tockenValied},
-          ),
-          data: data);
-      log('Response status code: ${response.statusCode}');
-      log('Response headers: ${response.headers}');
-      log('Response body: ${response.data}');
-      if (response.statusCode == 200) {
-        final userModel = AppUser.fromJson(response.data);
-        log(userModel.accessToken, name: "accssess token");
+      final response =
+          await api.general.post(ApiEndpoints.verifyOtpApi, data: requestData);
+      final userData = AppUser.fromJson(response.data);
```

Comment:
Remove accessToken variable and non-user related variables from UserModel class its should only contains user related data from api